### ISSUE TITLE
False positive with embedded types

### ISSUE THREAD

**Issue by scop** [junk1tm/musttag#12]
```go
package main

import (
	"encoding/json"
	"fmt"
)

type Foo struct {
	F string `json:"f"`
}

type Bar struct {
	Foo
	B int `json:"b"`
}

func main() {
	x, _ := json.Marshal(Bar{Foo{"foo"}, 42})
	fmt.Println(string(x))
}
```

With current musttag `@latest`:

```shellsession
$ musttag t.go 
/tmp/t.go:12:6: exported fields should be annotated with the "json" tag
```

**Comment by SVilgelm** [comment]
also faced this issue today

**Comment by junk1tm** [comment]
Nice catch, will fix that in the next release

**Comment by junk1tm** [comment]
@scop @SVilgelm please take a look at https://github.com/junk1tm/musttag/releases/tag/v0.4.4

**Comment by SVilgelm** [comment]
Thanks, I will check today

**Comment by scop** [comment]
Thanks, seems to fix the most glaring false positives.

However, it appears this was done by disabling checking embedded types entirely. I think it would be valuable to check them recursively instead. For structs under one's control this should not cause any problems, but embedding a third party and getting reports for that could be a little controversial.

If it was my call, I think I'd do the recursion without consideration where the embedded type originates from, the potential problem would be the same anyway. That could always be `//nolint`ed away (at least when using through golangci-lint), and I suppose embedding third party types is not as common as doing it within a project or a group of interrelated ones.

Anyway, #15 is definitely an improvement already :+1:

**Comment by junk1tm** [comment]
@scop thank you for the feedback, appreciate it!

If I get you right, you're talking about cases when an embedded struct contains some untagged fields. Well, the current implementation already handles it. If we take your example above and add a field with no tag, `musttag` will report the `Foo` type:

```go
package main

import (
	"encoding/json"
	"fmt"
)

type Foo struct {
	F     string `json:"f"`
	NoTag string
}

type Bar struct {
	Foo
	B int `json:"b"`
}

func main() {
	x, _ := json.Marshal(Bar{Foo{"foo", "-"}, 42})
	fmt.Println(string(x))
}
```

> main.go:8:6: exported fields should be annotated with the "json" tag

The following code, which was added in #15, reports fields that are untagged **and** not embedded.

 ```go
if _, ok := st.Lookup(tag); !ok {
	if !s.Field(i).Embedded() {
		return false
	}
}
```

So, if a field is an embedded struct, it will be checked recursively just like any other struct.

Speaking about 3rd party types and false positives, at first I was thinking about adding some config option, so one could disable checking of specific types, but later I decided that `//nolint` would do just fine (should probably support it natively too, outside of `golangci-lint`).

As of today, there is already a few issues about reports outside of the main module (see [this](https://github.com/italia/developers-italia-api/issues/190) and #17), so I think we should ignore 3rd party types by default. Although I'd prefer to keep cross package reports within the same module (not sure if it's possible to implement without module info in `*analysis.Pass` 🤔).
